modules/ting_smart_search/ting_smart_search.module

```diff
@@ -0,0 +1,661 @@
+      'description' => t('Allow role to configure smart search.'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ *
+ * It simply tells panels where to find the .inc files that define various
+ * args, contexts, content_types.
+ */
+function ting_smart_search_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'ctools' && !empty($plugin)) {
+    return "plugins/$plugin";
+  }
+}
+
+/**
+ * Implements hook_ctools_plugin_api().
+ */
+function ting_smart_search_ctools_plugin_api($module, $api) {
+  if ($module == 'page_manager' && $api == 'pages_default') {
+    return array('version' => 1);
+  }
+  if ($module == "strongarm" && $api == "strongarm") {
+    return array("version" => "1");
+  }
+}
+
+/**
+ * Implements hook_cron().
+ *
+ * Gets update off auto data every day and update of search data every week.
+ */
+function ting_smart_search_cron() {
+  watchdog('Smart Search', 'imported autodata time', array(), 'WATCHDOG_ERROR');
+  $last_run_auto = variable_get('ting_smart_search_last_cron_auto', 0);
+  $interval_auto = variable_get('ting_smart_search_interval_auto', 86400);
+  if (time() - $last_run_auto > $interval_auto) {
+    ting_smart_search_get_autodata_from_webtrekk();
+    watchdog('Smart Search', 'imported autodata: %time', array('%time' => time()), 'WATCHDOG_INFO');
+    variable_set('ting_smart_search_last_cron_auto', time());
+  }
+
+  $last_run_feed = variable_get('ting_smart_search_last_cron_feed', 0);
+  $interval_feed = variable_get('ting_smart_search_interval_feed', 604800);
+  if (time() - $last_run_feed > $interval_feed) {
+    ting_smart_search_get_searchdata_from_webtrekk();
+    watchdog('Smart Search', 'imported search data: %time', array('%time' => time()), 'WATCHDOG_INFO');
+    variable_set('ting_smart_search_last_cron_feed', time());
+  }
+}
+
+/**
+ * For use in the submit function of form
+ *
+ * Cache_clear_all(':'.current_path(), 'cache_ting', TRUE);
+ * Drupal_set_message('Cache for '.'*'.current_path().' this record cleared.');
+ *
+ * @param string $cid
+ *   The cache id.
+ */
+function ting_smart_search_opensearch_cache_key(&$cid) {
+  // Determine context.
+  $context = 'search';
+
+  // Modify cache key based on some context.
+  if ($context === 'search') {
+    $cid = ':' . current_path() . ':' . $cid;
+  }
+}
+
+/**
+ * Set additional parameters to ting request.
+ *
+ * @param object $request
+ *   The Ting request.
+ *
+ * @return null
+ *   Returns nothing.
+ */
+function ting_smart_search_opensearch_pre_execute($request) {
+
+// Alter the request based on the request class.
+  switch (get_class($request)) {
+    case 'TingClientSearchRequest':
+      $smart_search_check_sort = '';
+      $smart_search_check_boosts = array();
+      $smart_search_use_auto = true;
+
+      // Store the userDefinedBoost if any.
+      $smart_search_userdefined_boosts = $request->userDefinedBoost;
+      if (substr($request->getQuery(), 0, 1) === "(") {
```

Comment:
Was wondering: what is this check for?

Reply:
An alternative approach would be to leverage ting_field_search's `ting_field_search_get_context()`. It sets a static variable when search_result is rendered:

if (ting_field_search_get_context() === 'search_result') {
  // Alter request.
}